### PR TITLE
Issue 3659: Lazy-load images rendered via theme_image()

### LINKED ISSUE #3659: [SEO][WP] Add a lazy-loading option for image fields
**Description of the need**

Native `` lazy-loading is coming to the web! https://bit.ly/loading-attribute … `` defers offscreen images until the user scrolls near them. Shipping in Chrome ~75 https://bit.ly/loading-i2s 

This was one of the recommendations Google PageSpeed gave me for one of my Backdrop sites.

We should support this in Backdrop core!

---

References:
* https://css-tricks.com/native-lazy-loading/
* https://web.dev/native-lazy-loading/

---

Advocate: @klonos 

---

PR that adds a global setting under 'Performance': ~https://github.com/backdrop/backdrop/pull/3256~

PR that adds a setting for image fields: https://github.com/backdrop/backdrop/pull/3273

### PR DESCRIPTION
**Pull request by klonos**
Fixes https://github.com/backdrop/backdrop-issues/issues/3659

### CODE REVIEW

**Comment by klonos** [comment]
...ZenCI acting up again 😓

File: core/modules/layout/layout.theme.inc [hunk 1]
```diff
@@ -407,6 +407,9 @@ function template_preprocess_block(&$variables) {
     if (isset($block->settings['image_path'])) {
       $variables['attributes']['style'] = 'background-image:url(' . $block->settings['image_path'] . ');';
       $variables['classes'][] = 'block-hero-image';
+      if (config_get('system.core', 'lazy_load_images')) {
+        $variables['attributes']['loading'] = 'lazy';
```
Comment (indigoxela) [inline core/modules/layout/layout.theme.inc]: The loading attribute can not be used on `div` tags.

Comment (klonos) [reply core/modules/layout/layout.theme.inc]: You are right @indigoxela. Fixed.

File: core/modules/image/tests/image.test [hunk 2]
```diff
@@ -1937,7 +1940,7 @@ class ImageThemeFunctionWebTestCase extends BackdropWebTestCase {
       'fragment' => $fragment,
     );
     $rendered_element = render($element);
-    $expected_result = '<a href="#' . $fragment . '"><img src="' . $url . '" alt="" /></a>';
+    $expected_result = '<a href="#' . $fragment . '"><img src="' . $url . '" alt="" ' . $lazy_load . '/></a>';
```
Comment (indigoxela) [inline core/modules/image/tests/image.test:1943]: Just a suggestion for test enhancement, ignore if you disagree... :wink: 

There's currently no test yet for the config setting of lazy_load_images. In one of those tests you could config_set() the value to FALSE and then check if the attribute is *not* there. Just to have a check for whether the setting is respected.

Comment (klonos) [reply core/modules/image/tests/image.test:1943]: I agree, and I think that this has already been suggested by @hosef. The test should not be added here though, as this series of tests is specific to testing image styles. I'll add a separate test to explicitly test whether the attribute is set or not, depending on the config setting.

File: core/includes/theme.inc [hunk 3]
```diff
@@ -2006,8 +2006,29 @@ function template_preprocess_image(&$variables) {
...
+ *       'attributes' => $image_attributes,
+ *     );
+ *     $output = theme('image', $image_variables);
+ *   @endcode
  */
 function theme_image($variables) {
+  $lazy_load = config_get('system.core', 'lazy_load_images');
+  if ($lazy_load && !isset($variables['attributes']['loading'])) {
+    $variables['attributes']['loading'] = 'lazy';
+  }
```
Comment (quicksketch) [inline core/includes/theme.inc]: It seems like this code should go in `template_preprocess_image()` rather than `theme_image()`.

**Comment by klonos** [comment]
Closing this in favor of https://github.com/backdrop/backdrop/pull/3273 (setting on each individual image field)

**Pull request closed without merging**